native-gradle-plugin/src/main/java/org/graalvm/buildtools/gradle/tasks/BuildNativeImageTask.java

```diff
@@ -154,22 +156,40 @@ public void exec() {
         if (!executablePath.exists() && getGraalVMHome().isPresent()) {
             executablePath = Paths.get(getGraalVMHome().get()).resolve("bin").resolve(NATIVE_IMAGE_EXE).toFile();
         }
-        if (executablePath.exists()) {
-            logger.log("Using executable path: " + executablePath);
-            String executable = executablePath.getAbsolutePath();
-            File outputDir = getOutputDirectory().getAsFile().get();
-            if (outputDir.isDirectory() || outputDir.mkdirs()) {
-                getExecOperations().exec(spec -> {
-                    spec.setWorkingDir(getWorkingDirectory());
-                    spec.args(args);
-                    getService().get();
-                    spec.setExecutable(executable);
+
+        try {
+            if (!executablePath.exists()) {
+                logger.log("Native Image executable wasn't found. We will now try to download it. ");
+                File graalVmHomeGuess = executablePath.getParentFile();
+
+                if (!graalVmHomeGuess.toPath().resolve(GU_EXE).toFile().exists()) {
+                    throw new GradleException("'" + GU_EXE + "' tool wasn't found. Is this JDK GraalVM distribution?");
```

Comment:
I would rephrase to something like "tool wasn't found. This probably means that JDK at <path> isn't a GraalVM distribution."